packages/web3-eth-iban/src/index.ts

```diff
@@ -0,0 +1,162 @@
+import { toChecksumAddress, isAddress, leftPad, hexToNumber } from 'web3-utils';
+import { IbanLengthError } from 'web3-common';
+import { IbanOptions } from './types';
+
+/**
+ * Prepare an IBAN for mod 97 computation by moving the first 4 chars to the end and transforming the letters to
+ * numbers (A = 10, B = 11, ..., Z = 35), as specified in ISO13616.
+ */
+const _iso13616Prepare = (iban: string): string => {
+	const A = 'A'.charCodeAt(0);
+	const Z = 'Z'.charCodeAt(0);
+
+	const upperIban = iban.toUpperCase();
+	const modifiedIban = `${upperIban.substr(4)}${upperIban.substr(0, 4)}`;
+
+	return modifiedIban
+		.split('')
+		.map(n => {
+			const code = n.charCodeAt(0);
+			if (code >= A && code <= Z) {
+				// A = 10, B = 11, ... Z = 35
+				return code - A + 10;
+			}
+			return n;
+		})
+		.join('');
+};
+
+/**
+ * return the bigint of the given string with the specified base
+ */
+const _parseInt = (str: string, base: number): bigint =>
+	[...str].reduce((acc, curr) => BigInt(parseInt(curr, base)) + BigInt(base) * acc, 0n);
+
+/**
+ * Calculates the MOD 97 10 of the passed IBAN as specified in ISO7064.
+ */
+const _mod9710 = (iban: string): number => {
+	let remainder = iban;
+	let block;
+
+	while (remainder.length > 2) {
+		block = remainder.slice(0, 9);
+		remainder = `${(parseInt(block, 10) % 97).toString()}${remainder.slice(block.length)}`;
+	}
+
+	return parseInt(remainder, 10) % 97;
+};
+
+const _isValid = (iban: string): boolean =>
+	/^XE[0-9]{2}(ETH[0-9A-Z]{13}|[0-9A-Z]{30,31})$/.test(iban) &&
+	_mod9710(_iso13616Prepare(iban)) === 1;
+
+export class Iban {
+	private readonly _iban: string;
+
+	public constructor(iban: string) {
+		if (_isValid(iban)) {
+			// error
+		}
+		this._iban = iban;
+	}
+
+	/**
+	 * check if iban number is direct
+	 */
+	private isDirect() {
+		return this._iban.length === 34 || this._iban.length === 35;
+	}
+
+	/**
+	 * Get the clients direct address from iban
+	 */
+	public toAddress = (): string => {
+		if (this.isDirect()) {
+			// check if Iban can be converted to an address
+			const base36 = this._iban.substr(4);
+			const bigInt = _parseInt(base36, 36); // convert the base36 string to a bigint
+			const paddedBigInt = leftPad(bigInt, 40);
+			return toChecksumAddress(paddedBigInt);
+		}
+		throw new IbanLengthError();
+	};
+	/**
+	 * This method should be used to create an ethereum address from a direct iban address
+	 */
+	public static toAddress = (iban: string): string => {
+		const ibanObject = new Iban(iban);
+		if (!ibanObject.isDirect()) {
```

Comment:
This `.isDirect()` is already done inside non static toAddress() , so i think its duplicate and we can remove it here.